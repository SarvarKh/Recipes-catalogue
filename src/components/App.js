import { indexOf } from "lodash";
import React, { useState, useEffect } from "react";
import { fetchMeals } from "../actions";
import { connect } from 'react-redux';

function App({fetchMeals}) {
  const [selectedCategory, setSelectedCategory] = useState('Seafood');

  const [meals, setMeals] = useState([]);
  useEffect(() => {
    fetchMeals();
  });

  const [categories, setCategories] = useState([]);
  useEffect(() => {
    fetch("https://www.themealdb.com/api/json/v1/1/list.php?c=list")
    .then(res => res.json())
    .then(data => setCategories(data.meals));
  }, []);

  const handleFilter = (e) => {
    setSelectedCategory(e);
  }

  return (
    <div>
      <h1>Categories</h1>
      <select onChange={(e) => handleFilter(e.target.value)} className="nav-item appearance-none cursor-pointer">
        {categories.map(category => (
          <option key={categories.indexOf(category)} value={category.strCategory}>
            {category.strCategory}
          </option>
        ))}
      </select>

      <h1>Meals</h1>
      {
        meals.map(meal => (
          <div key={meal.idMeal}>
            <h3>{meal.strMeal}</h3>
            <div className="img-container">
              <img src={meal.strMealThumb} />
            </div>
          </div>
        ))
      }
    </div>
  )
}

export default connect(null, { fetchMeals })(App);
