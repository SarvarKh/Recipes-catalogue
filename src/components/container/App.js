import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
  fetchCategories,
  fetchMeals,
  fetchDetailMeal,
  fetchByIngridient,
  fetchByArea,
} from '../../actions/index';
import Categories from '../presentation/Categories';
import Meals from '../presentation/Meals';

function App({
  fetchMeals,
  fetchCategories,
  fetchDetailMeal,
  fetchByIngridient,
  fetchByArea,
  meals,
}) {
  useEffect(() => {
    fetchMeals('Beef');
    fetchCategories();
  }, []);

  const handleClick = (e) => {
    fetchMeals(e);
  };

  const clickOnDetailMeal = (e) => {
    fetchDetailMeal(e);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const inputValue = e.target[0].value;

    fetchByIngridient(inputValue);
    fetchByArea(inputValue);
  };

  return (
    <main>
      <form onSubmit={(e) => handleSubmit(e)}>
        <input placeholder="Search meal by main ingredient (e.g. Chicken Breast) and area (e.g. American, Canadian)" name="by-cat-and-area" />
        <button type="submit">
          <i className="fas fa-search" />
        </button>
      </form>

      <Categories handleClick={handleClick} />
      <Meals meals={meals} clickOnDetailMeal={clickOnDetailMeal} />
    </main>
  );
}

const mapStateToProps = (state) => ({
  meals: state.meals.meals,
});

export default connect(mapStateToProps, {
  fetchMeals, fetchCategories, fetchDetailMeal, fetchByIngridient, fetchByArea,
})(App);

App.propTypes = {
  meals: PropTypes.instanceOf(Array).isRequired,
  fetchMeals: PropTypes.func.isRequired,
  fetchCategories: PropTypes.func.isRequired,
  fetchDetailMeal: PropTypes.func.isRequired,
  fetchByIngridient: PropTypes.func.isRequired,
  fetchByArea: PropTypes.func.isRequired,
};
