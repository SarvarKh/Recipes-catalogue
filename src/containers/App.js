import { indexOf } from "lodash";
import React, { useState, useEffect } from "react";

function App() {
  const [meals, setMeals] = useState([]);
  const [categories, setCategories] = useState([]);

  useEffect(() => {
    fetch("https://www.themealdb.com/api/json/v1/1/filter.php?c=Seafood")
    .then(res => res.json())
    .then(data => setMeals(data.meals));

    fetch("https://www.themealdb.com/api/json/v1/1/list.php?c=list")
    .then(res => res.json())
    .then(data => setCategories(data.meals));
  }, [])

    return (
      <div>
        <h1>Categories</h1>
          <select className="nav-item appearance-none cursor-pointer">
            {categories.map(category => (
              <option key={indexOf(category)} value={category.strCategory}>
                {category.strCategory}
              </option>
            ))}
          </select>


        <h1>Meals</h1>
        {
          meals.map(meal => (
            <div key={meal.idMeal}>
              <h3>{meal.strMeal}</h3>
              <div className="img-container">
                <img src={meal.strMealThumb} />
              </div>
            </div>
          ))
        }
      </div>
    )
}

export default App;
