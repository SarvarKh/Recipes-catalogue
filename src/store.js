import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './reducers/rootReducer';

const initialState = {
  detailMeal: {},
  meals: [],
  categories: [],
  searchedItems: [],
};

const middleware = [thunk];

const store = createStore(
  rootReducer,
  initialState,
  compose(
    applyMiddleware(...middleware),
  ),
);

export default store;
