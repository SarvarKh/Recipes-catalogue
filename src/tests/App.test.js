import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../store';
import App from '../components/container/App';

it('renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><App /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});