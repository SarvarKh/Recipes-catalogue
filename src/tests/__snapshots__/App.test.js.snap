// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main>
  <form
    onSubmit={[Function]}
  >
    <input
      name="by-cat-and-area"
      placeholder="Search meal by main ingredient (e.g. Chicken Breast) and area (e.g. American, Canadian)"
    />
    <button
      type="submit"
    >
      <i
        className="fas fa-search"
      />
    </button>
  </form>
  <h1
    className="titles"
  >
    Categories
  </h1>
  <div
    className="flex categories-container"
  >
    <h2>
      Loading...
    </h2>
  </div>
  <h1
    className="titles t-meal"
  >
    Meals
  </h1>
  <div
    className="flex meal-container"
  >
    <h2>
      Loading...
    </h2>
  </div>
</main>
`;
